types/src/on_chain_config/execution_config.rs

```diff
@@ -30,12 +30,22 @@ impl OnChainExecutionConfig {
     }
 
     /// The per-block gas limit being used.
-    pub fn block_gas_limit(&self) -> Option<u64> {
+    pub fn block_gas_limit_type(&self) -> BlockGasLimitType {
         match &self {
-            OnChainExecutionConfig::Missing => None,
-            OnChainExecutionConfig::V1(_config) => None,
-            OnChainExecutionConfig::V2(config) => config.block_gas_limit,
-            OnChainExecutionConfig::V3(config) => config.block_gas_limit,
+            OnChainExecutionConfig::Missing => BlockGasLimitType::NoLimit,
+            OnChainExecutionConfig::V1(_config) => BlockGasLimitType::NoLimit,
+            OnChainExecutionConfig::V2(config) => config
+                .block_gas_limit
+                .map_or(BlockGasLimitType::NoLimit, BlockGasLimitType::Limit),
+            OnChainExecutionConfig::V3(config) => config
+                .block_gas_limit
+                .map_or(BlockGasLimitType::NoLimit, BlockGasLimitType::Limit),
+        }
+    }
+
+    pub fn block_executor_onchain_config(&self) -> BlockExecutorOnchainConfig {
```

Comment:
should we move the config onchain as well? or there's some pieces that's local?

Reply:
you mean BlockExecutorOnchainConfig to be in the onchain config? 

I don't want that - as onchain config itself needs to have all versioning and stuff. I want to have one thing representing the value, that I can change without issues, that translates all different onchain versions into

Reply:
it's a little confusing to have OnChainConfig not actually part of the onchain config 😄

Reply:
BlockExecutorValuesFromOnchainConfig or something?

do you have a better name?

Reply:
I don't have a better name, probably just add a doc comment for the struct that it's derived from on chain values